tests/python/relax/test_relax_tensor_ops.py

```diff
@@ -49,6 +50,13 @@ def relax_build_and_run(f, inputs):
         return vm["default"](*inputs)
 
 
+def run_relax_module(module, *input):
+    mod = OperatorLegalizer(module).transform()
+    ex = relax.vm.build(mod, tvm.target.Target("llvm"))
+    vm = relax.VirtualMachine(ex, tvm.cpu())
+    return vm["main"](*input)
+
+
```

Comment:
No need to legalize, build and then run here. It’s enough to test the structural equality. It is simple and “unit”.

Reply:
But shouldn't we test the correctness of op legalization part? 🤔

Reply:
Yes we should, but not here.

The entire pipeline contains several phases:
* P1. high-level operator creation,
* P2. legalization, convert high-level operators to CallTIRs with corresponding PrimFuncs
* P3. build and run the IRModule (meanwhile build and run the PrimFuncs).

These phases are tested separately. Here we only test the creation. P2 is tested in `test_op_legalizer.py`, and P3 is a part of TOPI test coverage.

Reply:
Just noted that we didn’t add the tests of P2 into `test_op_legalizer.py`. Please kindly do so :-)

Also please check if we have test coverage for `topi.collapse_sum` on TOPI side.

Reply:
Got it.
I still have a question: lower + build + run + assert_allclose is a end-to-end way to test the correctness of operators. That is convincing and straightforward. 
But when seperating test into these phases, the correctness seems not that clear?

Reply:
Here we want our tests as unit as possible, so that in case of any error, these unit tests can serve as the some minimal reproducible code. As long as we make sure each phase has good test coverage then it is fine.

Reply:
Okay!